edgexfoundry/app-functions-sdk-go
# refactor: feat: Replace gorilla/mux router with echo

closes #1436







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

      <link to docs PR>

## Testing Instructions
run EdgeX non-secure stack
build and run app-template
send GET commands to the following
- localhost:59740/api/v3/ping
- localhost:59740/api/v3/version
- localhost:59740/api/v3/config
- localhost:59740/api/v3/hello
Verify above return 200 status with expected responses
send POST to localhost:59740/api/v3/secret with
```
{
   "apiVersion":"v3",
   "secretName":"credentials",
   "secretData":[
      {
         "Key" :"username",
         "Value":"user"
      },
      {
         "Key" :"password",
         "Value":"pwd"
      }
   ]
}
```
Verify returns 201 status

## New Dependency Instructions (If applicable)
